File: tests/ForExpJS.hs
module ForExpJS where

foreign export javascript addOne :: Int -> IO Int
foreign export javascript "isPos" gt0 :: Int -> Bool
foreign export javascript scale :: Double -> Double -> Double

addOne :: Int -> IO Int
addOne x = do
  putStrLn "addOne called"
  pure (x + 1)

gt0 :: Int -> Bool
gt0 x = x > 0

scale :: Double -> Double -> Double
scale = (*)

File: tests/ForExpJS.mjs
// Driver for testforexpjs: calls the EMSCRIPTEN_KEEPALIVE exports of the
// emcc-compiled ForExpJS module (fejs.mjs, built by the Makefile rule).
const FEJS = (await import('./fejs.mjs')).default;
const m = await FEJS();
m._mhs_init();
console.log(m._addOne(41));
console.log(m._isPos(3), m._isPos(-3));
console.log(m._scale(2.5, 4));
